src/zasm/src/serialization/serializer.cpp

```diff
@@ -128,18 +128,10 @@ namespace zasm
             alignBytesMissing -= dataIndex;
         }
 
-        if (nodeEntry.length != 0)
+        if (nodeEntry.length != 0 && !ctx.needsExtraPass
```

Comment:
`!ctx.needsExtraPass` can be removed here, that's just an extra branch to avoid setting it again.

Reply:
Okey,Done

Reply:
Done